### PR TITLE
Use try-catch structure in `do_upgrade.ps1` script

### PR DESCRIPTION
**Pull request by mjcr99**
## Description

Hi team,

this PR introduces the try-catch structure to try stopping the `win32ui` and the `Wazuh` service in the `do_upgrade.ps1 `script used in the WPK upgrade procedure.

Using this structure, we ensure we will properly catch any problem related to killing the mentioned process and service, which seem to have failed in a corner case.

## Tests

:green_circle: [Packages - Build Wazuh agent Windows #200](https://github.com/wazuh/wazuh-agent-packages/actions/runs/18281790128/job/52053373038)

:green_circle: [Packages - Build WPK #89](https://github.com/wazuh/wazuh-agent-packages/actions/runs/18284412035)

## Review Checklist

- [x] Code changes reviewed
- [x] Relevant evidence provided
- [x] Tests cover the new functionality
- [ ] Configuration changes documented
- [ ] Developer documentation reflects the changes
- [x] Meets requirements and/or definition of done
- [x] No unresolved dependencies with other issues

### CODE REVIEW

File: src/win32/do_upgrade.ps1 [hunk 1]
```diff
@@ -173,22 +173,37 @@ function Get-MSIProductVersion {
...
+
+    # Try to stop Wazuh service
+    try {
+        Write-Output "$(Get-Date -format u) - Stopping Wazuh service." >> .\upgrade\upgrade.log
+        Stop-Service -Name "Wazuh" -Force -ErrorAction Stop
+    } catch {
+        Write-Output "$(Get-Date -format u) - Tried to stop Wazuh service: $($_.Exception.Message)" >> .\upgrade\upgrade.log
+    }
+
+    Start-Sleep -Seconds 5
```
Comment (jpcerrone) [inline src/win32/do_upgrade.ps1:193]: Might be worth it to add a comment explaining why the hard wait is needed

Comment (mjcr99) [reply src/win32/do_upgrade.ps1:193]: Addressed with: https://github.com/wazuh/wazuh/pull/32452/commits/4f414a3bf57d5d74d82986864612c6d1192eaa4f

File: src/win32/do_upgrade.ps1 [hunk 2]
```diff
@@ -221,7 +236,15 @@ if ($msi_new_version -ne $null) {
 
 
 # Ensure no other instance of msiexec is running by stopping them
-Get-Process msiexec | Stop-Process -ErrorAction SilentlyContinue -Force
+try {
+    $proc = Get-Process -Name "msiexec" -ErrorAction Stop
+    if ($proc) {
+        Stop-Process -InputObject $proc -Force -ErrorAction Stop
+        Write-Output "$(Get-Date -Format u) - Killed msiexec process(es)." >> .\upgrade\upgrade.log
+    }
```
[the remark this answers is not in the document]
Comment (mjcr99) [reply src/win32/do_upgrade.ps1]: Addressed with: https://github.com/wazuh/wazuh/commit/4f414a3bf57d5d74d82986864612c6d1192eaa4f

**Comment by mjcr99** [comment]
After the last changes:

:green_circle: [Packages - Build Wazuh agent Windows #200](https://github.com/wazuh/wazuh-agent-packages/actions/runs/18281790128/job/52129322918)

:green_circle: [Packages - Build WPK #89](https://github.com/wazuh/wazuh-agent-packages/actions/runs/18308531203)

File: src/win32/do_upgrade.ps1 [hunk 3]
```diff
@@ -173,22 +173,38 @@ function Get-MSIProductVersion {
...
+    # Try to stop Wazuh service
+    try {
+        Write-Output "$(Get-Date -format u) - Stopping Wazuh service." >> .\upgrade\upgrade.log
+        Stop-Service -Name "Wazuh" -Force -ErrorAction Stop
+    } catch {
+        Write-Output "$(Get-Date -format u) - Tried to stop Wazuh service: $($_.Exception.Message)" >> .\upgrade\upgrade.log
+    }
+
+    # Wait for Wazuh service to fully stop
+    Start-Sleep -Seconds 5
```
[the remark this answers is not in the document]
Comment (mjcr99) [reply src/win32/do_upgrade.ps1:193]: It's indicated.

File: src/win32/do_upgrade.ps1 [hunk 4]
```diff
@@ -173,22 +173,38 @@ function Get-MSIProductVersion {
 
 # Stop UI and launch the MSI installer
 function install {
-    kill -processname win32ui -ErrorAction SilentlyContinue -Force
-    Stop-Service -Name "Wazuh"
+    # Try to stop win32ui
+    try {
+        Write-Output "$(Get-Date -format u) - Stopping win32ui process." >> .\upgrade\upgrade.log
+        Stop-Process -Name "win32ui" -Force -ErrorAction Stop
```
[the remark this answers is not in the document]
Comment (mjcr99) [reply src/win32/do_upgrade.ps1:179]: Not necessary.

**Review by jpcerrone — APPROVED** [review]
LGTM
